OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -96,6 +119,103 @@ public void onPageLoadProgressChanged(WebView webView, int progress) {
         }
     }
 
+    private void fetchDateBannerInfo() {
+        if (canFetchBannerInfo && unit.getType() == BlockType.PROBLEM) {
+            courseDateViewModel = new ViewModelProvider(this, new ViewModelFactory()).get(CourseDateViewModel.class);
+            initObserver();
+            courseDateViewModel.fetchCourseDatesBannerInfo(unit.getCourseId(), true);
+        }
+    }
+
+    private void initInfoBanner(CourseBannerInfoModel courseBannerInfo) {
+        if (courseBannerInfo == null) {
+            infoBanner.setVisibility(View.GONE);
+            return;
+        }
+        TextView textView = (TextView) infoBanner.findViewById(R.id.banner_info);
+        Button button = (Button) infoBanner.findViewById(R.id.btn_shift_dates);
+        String buttonText = "";
+        // Currently we are only handling RESET_DATES case,
+        // TODO UPGRADE_TO_GRADED & UPGRADE_TO_RESET will be enable once we are allowed to do payment through mobile
+        switch (courseBannerInfo.getDatesBannerInfo().getCourseBannerType()) {
+            case UPGRADE_TO_GRADED:
+                textView.setText(getText(R.string.course_dates_banner_upgrade_to_graded));
+                break;
+            case UPGRADE_TO_RESET:
+                textView.setText(getText(R.string.course_dates_banner_upgrade_to_reset));
+                break;
+            case RESET_DATES:
+                textView.setText(getText(R.string.course_dates_banner_reset_date));
+                buttonText = getContextOrThrow().getString(R.string.course_dates_banner_reset_date_button);
+                break;
+            case INFO_BANNER:
+                textView.setText(getText(R.string.course_dates_info_banner));
+                break;
+            case BLANK:
+                infoBanner.setVisibility(View.GONE);
+                break;
+        }
+
+        if (!TextUtils.isEmpty(textView.getText())) {
+            if (!TextUtils.isEmpty(buttonText)) {
+                button.setText(buttonText);
+                button.setVisibility(View.VISIBLE);
+                button.setOnClickListener(v -> {
+                    courseDateViewModel.resetCourseDatesBanner(unit.getCourseId());
+                });
+            }
+            infoBanner.setVisibility(View.VISIBLE);
+        }
```

Comment:
also need to check the `buttonText` is empty, cuz button should be hide after shifting the dates. 

```
        if (!TextUtils.isEmpty(textView.getText())) {
            if (!TextUtils.isEmpty(buttonText)) {
                button.setText(buttonText);
                button.setOnClickListener(v -> {
                    courseDateViewModel.resetCourseDatesBanner(unit.getCourseId());
                });
            }
            infoBanner.setVisibility(View.VISIBLE);
        }
        button.setVisibility(TextUtils.isEmpty(buttonText) ? View.GONE : View.VISIBLE);
```

Reply:
by default button is hidden, and on the response of the date shifted API, banner info API also called so don't need to handle this case